src/MarginTrading.AzureRepositories/Entities/OvernightSwapHistoryEntity.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Collections.Generic;
+using MarginTrading.Backend.Core;
+using Microsoft.WindowsAzure.Storage.Table;
+using Newtonsoft.Json;
+
+namespace MarginTrading.AzureRepositories.Entities
+{
+	public class OvernightSwapHistoryEntity : TableEntity, IOvernightSwapHistory
+	{
+		public string AccountId { get; set; }
+		public string Instrument { get; set; }
+		public string Direction { get; set; }
+		OrderDirection? IOvernightSwapState.Direction => 
+			Enum.TryParse<OrderDirection>(Direction, out var direction) ? direction : (OrderDirection?)null;
+		public DateTime Time { get; set; }
+		public string OpenOrderIds { get; set; }
+		List<string> IOvernightSwapState.OpenOrderIds => JsonConvert.DeserializeObject<List<string>>(OpenOrderIds);
+		public double Value { get; set; }
+		decimal IOvernightSwapState.Value => (decimal) Value;
+		public double SwapRate { get; set; }
+		decimal IOvernightSwapState.SwapRate => (decimal) SwapRate;
+		
+		public bool IsSuccess { get; set; }
+		public string Exception { get; set; }
+		Exception IOvernightSwapHistory.Exception => JsonConvert.DeserializeObject<Exception>(Exception);
+		
+		public static string GetKey(string accountId, string instrument, OrderDirection? direction) =>
+			$"{accountId}_{instrument ?? ""}_{direction?.ToString() ?? ""}";
+
+		public static OvernightSwapHistoryEntity Create(IOvernightSwapHistory obj)
+		{
+			return new OvernightSwapHistoryEntity
+			{
+				PartitionKey = obj.AccountId,
+				RowKey = GetKey(obj.AccountId, obj.Instrument, obj.Direction),
```

Comment:
PartitionKey + RowKey does not depend on date and not good idea for history. I know that in repo you generate PK, but it should be removed from here.
Copy-paste? )

Reply:
That was copy-paste, yep. Fixed.